src/main/java/org/websoso/WSSServer/auth/validator/NotificationAuthorizationValidator.java

```diff
@@ -0,0 +1,49 @@
+package org.websoso.WSSServer.auth.validator;
+
+import static org.websoso.WSSServer.domain.common.Role.ADMIN;
+import static org.websoso.WSSServer.exception.error.CustomNotificationError.NOTIFICATION_ADMIN_ONLY;
+import static org.websoso.WSSServer.exception.error.CustomNotificationError.NOTIFICATION_NOT_FOUND;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import org.websoso.WSSServer.domain.Notification;
+import org.websoso.WSSServer.domain.User;
+import org.websoso.WSSServer.exception.exception.CustomNotificationException;
+import org.websoso.WSSServer.repository.NotificationRepository;
+
+@Component
+@RequiredArgsConstructor
+public class NotificationAuthorizationValidator implements ResourceAuthorizationValidator {
+
+    private final NotificationRepository notificationRepository;
+
+    @Override
+    public boolean hasPermission(Long resourceId, User user) {
+        if (resourceId == null) {
+            return isAdmin(user);
+        }
+
+        // 수정이나 삭제에서 리소스 존재 여부 검증을 위함
+        Notification notification = getNotificationOrException(resourceId);
+        return isAdmin(user);
```

Comment:
p5; 이렇게 로직을 수정하는건 어떤가요?
조건절이 참이 되도록 하는 것도 좋지만, 지금과 같은 경우에는 리턴값이 두 경우 모두 같아서, 조건절을 참으로 하는게 가독성을 해치는 것 같습니다.
```suggestion
@Override
public boolean hasPermission(Long resourceId, User user) {
    if (resourceId != null) {
        getNotificationOrException(resourceId);
    }
    return isAdmin(user);
}
```

Reply:
제시해주신 코드가 좀 더 간결하고 코드 중복이 사라진다는 점이 있긴 한데, 의도를 살리기 위해서 저렇게 구현했어요.
POST 요청의 경우 resourceId가 null이기 때문에 리소스 존재여부를 검증하지 않고(로직 스킵) 관리자여부만 확인하고,
PUT/DELETE 요청의 경우 리소스 존재 여부 확인 후 관리자 여부를 확인하도록이요.

어떻게 생각하시나요~

Reply:
처음에 코드를 봤을 때는 리턴값이 같은데, 두번 쓰여 있어 한번에 이해가 안되나 했었는데
지금보니 validate역할을 하는 함수가 getNotificationOrException()으로 쓰여 있어서 한번에 읽히지 않는게 큰 것 같네요.
그래도 제 눈에는 `if (resourceId != null)`가 더 잘 읽히긴 하는데 이건 사람마다 다를거라고 생각하고,
아직까지는 get~OrException()이 좋을지, validate 함수를 따로 쓰는게 좋을지 잘 모르겠어서... 이 부분은 추후 notification 수정 및 삭제 pr에서 다시 이야기해보는건 어떨까요? 더 궁금하신 거 없으시면 읽고 resolve 하셔두 됩니다!!